### PR TITLE
Stop serving endpointslice v1beta1

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Review by hanxiaop — APPROVED** [review]
Release note looks good

**Comment by hanxiaop** [comment]
It appears that `v1beta1` is no longer being served as of version 1.25. Should we continue to support it until our minimum supported k8s version meets it?

**Review by howardjohn — COMMENTED** [review]
This is done in https://github.com/istio/istio/pull/43725 as well, there is a lot more legacy code to cleanup related to endpointslices v1beta1 that I took care of in that PR

**Comment by hzxuzhonghu** [comment]
Should split unrelated code out from https://github.com/istio/istio/pull/43725?

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -261,13 +261,13 @@ var (
 		"If enabled, Pilot will keep track of old versions of distributed config for this duration.",
 	).Get()
 
-	enableEndpointSliceController, endpointSliceControllerSpecified = env.Register(
+	EnableEndpointSliceController = env.Register(
 		"PILOT_USE_ENDPOINT_SLICE",
-		false,
+		true,
```
Comment (howardjohn) [inline pilot/pkg/features/pilot.go:266]: Why change this? we can just use Endpoints on older than v1.21 and EndpointSlice on 1.21+.

Then this change just removes the ability to use Endpointslice on v1.20 which is not a big deal. As is, this PR breaks all users on 1.20, and we are aiming to support it for at least another release.

**Review by howardjohn — COMMENTED** [review]
I opened https://github.com/istio/istio/pull/43832 with the changes from https://github.com/istio/istio/pull/43725 split out. I am ok with moving forward with this as well but thought it would be easier to show what I meant in a PR instead of comments

**Pull request closed without merging**